src/main/kotlin/live/adamlearns/animalroyale/TwitchChat.kt

```diff
@@ -368,5 +331,66 @@ class TwitchChat(private val gameContext: GameContext) {
 
     companion object {
         val ADMIN_PERMISSIONS = listOf(CommandPermission.OWNER, CommandPermission.BROADCASTER, CommandPermission.MODERATOR)
+
+        private var cachedClosestColors: MutableMap<String, DyeColor> = mutableMapOf()
+
+        private fun getDyeColorFromHex(hex: String): DyeColor =
+            // These are the 'default' Twitch colors. For custom colors, we try to find the closest one.
+            when (hex) {
+                // rgb(255, 0, 0)
+                "#FF0000" -> DyeColor.RED
+                // rgb(178, 34, 34)
+                "#B22222" -> DyeColor.RED
+                // rgb(0, 128, 0)
+                "#008000" -> DyeColor.GREEN
+                // rgb(154, 205, 50)
+                "#91CD32" -> DyeColor.GREEN
+                // rgb(46, 139, 87)
+                "#2E8B57" -> DyeColor.GREEN
+                // rgb(0, 255, 127)
+                "#00FF7F" -> DyeColor.GREEN
+                // rgb(0, 0, 255)
+                "#0000FF" -> DyeColor.BLUE
+                // rgb(30, 144, 255)
+                "#1E90FF" -> DyeColor.BLUE
+                // rgb(95, 158, 160)
+                "#5F9EA0" -> DyeColor.LIGHT_BLUE
+                // rgb(255, 127, 80)
+                "#FF7F50" -> DyeColor.ORANGE
+                // rgb(255, 69, 0)
+                "#FF4500" -> DyeColor.ORANGE
+                // rgb(210, 105, 30)
+                "#D2691E" -> DyeColor.ORANGE
+                // rgb(218, 165, 32)
+                "#DAA520" -> DyeColor.YELLOW
+                // rgb(255, 105, 180)
+                "#FF69B4" -> DyeColor.PINK
+                // rgb(138, 43, 226)
+                "#8A2BE2" -> DyeColor.PURPLE
```

Comment:
Why not just _always_ use `getClosestDyeColorFromHex`? I recognize that the colors may not be as close as they could be by directly choosing them here, but if they're not drastically different, then I think it would be a good change to make.

Reply:
I just thought it'd be slightly more optimal to have this for the most common/known colors, but the end result would (should) probably be the same.

I can check out whether these all match by using `getClosestDyeColorFromHex` and if they all do we can go with that 👍🏽

Reply:
So I did end up verifying this, and these were the mismatches:

![image](https://github.com/user-attachments/assets/464ecd30-921b-4150-a5ef-2397b446ddd3)

So for some of them it found better options than I had, but others it botches. If it's ok with you I'll fix the ones I got wrong but still leave these predefined ones.

Reply:
That sounds good to me! 👍